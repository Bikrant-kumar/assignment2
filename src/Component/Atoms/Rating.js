import React from 'react';
import Rating from '@material-ui/lab/Rating';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    '& > * + *': {
      marginTop: theme.spacing(1),
    },
  },
}));

export default function HalfRating(props) {
  const classes = useStyles();
  function handleInputChange(event) {
    props.getValue(event.target.value);
  }

  return (
    <div className={classes.root}>
      <Rating name="half-rating-read" size="large" defaultValue={props.defaultValue || ""} precision={0.5} readOnly={props.readOnly} 
      onChange={handleInputChange}
      data-testid="checkRating"
      />
    </div>
  );
}
