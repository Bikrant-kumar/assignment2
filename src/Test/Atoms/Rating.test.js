import React from 'react';
import { screen,render } from '@testing-library/react';
import Rating from '../../Component/Atoms/Rating';
import renderer from 'react-test-renderer';


describe(Rating ,() => {

   test('Rating check',() => {
      render(<Rating defaultValue={3}/> );
      expect(screen.getByTestId("checkRating")).toBeInTheDocument();
  });

   test('test Rating with snapshot',()=>{
      const component= renderer.create(<Rating defaultValue={4.5}/>).toJSON();
      expect(component).toMatchSnapshot();
   });

})

