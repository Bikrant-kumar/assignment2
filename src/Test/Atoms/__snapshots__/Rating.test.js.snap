// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HalfRating test Rating with snapshot 1`] = `
<div
  className="makeStyles-root-2"
>
  <span
    aria-label={null}
    className="MuiRating-root MuiRating-sizeLarge"
    data-testid="checkRating"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    role={null}
  >
    <span
      className="MuiRating-decimal"
    >
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-0-5"
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          0.5 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-0-5"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={0.5}
      />
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-1"
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          1 Star
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-1"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={1}
      />
    </span>
    <span
      className="MuiRating-decimal"
    >
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-1-5"
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          1.5 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-1-5"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={1.5}
      />
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-2"
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          2 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-2"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={2}
      />
    </span>
    <span
      className="MuiRating-decimal"
    >
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-2-5"
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          2.5 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-2-5"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={2.5}
      />
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-3"
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          3 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-3"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={3}
      />
    </span>
    <span
      className="MuiRating-decimal"
    >
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-3-5"
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          3.5 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-3-5"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={3.5}
      />
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-4"
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          4 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-4"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={4}
      />
    </span>
    <span
      className="MuiRating-decimal"
    >
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-4-5"
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "50%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          4.5 Stars
        </span>
      </label>
      <input
        checked={true}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-4-5"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={4.5}
      />
      <label
        className="MuiRating-label"
        htmlFor="half-rating-read-5"
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyhidden"
        >
          5 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyhidden"
        disabled={false}
        id="half-rating-read-5"
        name="half-rating-read"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={5}
      />
    </span>
  </span>
</div>
`;
